creator-node/src/snapbackSM/snapbackSM.js

```diff
@@ -274,15 +293,75 @@ class SnapbackSM {
   }
 
   /**
-   * Submit POA:UserReplicaSetFactory.updateReplicaSet() transaction to move user off unhealthy node
+   * Depending on the size of `unhealthyReplicas`:
+   * 1. Determine a new replica set
+   * 2. Write new replica set to URSM
+   * 3. Sync data to new replica set
    *
-   * @dev TODO use libs for CreatorNodeSelection logic + relayed chain call
-   * @dev One issue - since it doesn't actually rectify broken replica sets, this will log every time it is processed
-   */
-  async issueUpdateReplicaSetOp (userId, wallet, primary, secondary1, secondary2, unhealthyReplica) {
-    // await libs.userReplicaSetManagerClient._updateReplicaSet(userId, primary, [new secondaries], primary, [old secondaries])
+   * @param {number} userId user id to issue a reconfiguration for
+   * @param {string} wallet wallet address of user id
+   * @param {string} primary endpoint of the current primary node on replica set
+   * @param {string} secondary1 endpoint of the current first secondary node on replica set
+   * @param {string} secondary2 endpoint of the current second secondary node on replica set
+   * @param {string[]} unhealthyReplicas array of endpoints of current replica set nodes that are unhealthy
+   * @param {string[]} healthyPeers array of endpoints of current peer set nodes slice that are healthy
+  */
+  async issueUpdateReplicaSetOp (userId, wallet, primary, secondary1, secondary2, unhealthyReplicas, healthyPeers) {
+    this.log(`[issueUpdateReplicaSetOp] userId=${userId} wallet=${wallet} unhealthy replica set=[${unhealthyReplicas}] | current healthy peer set=[${healthyPeers}]`)
+
+    // TODO: make optimizely flag
+    if (!this.snapbackReconfigEnabled) return
+
+    const unhealthyReplicasSet = new Set(unhealthyReplicas)
+    const baseSyncRequestParams = { userWallet: wallet, syncType: SyncType.Manual }
+    let newReplicaSetSPIds = []
+
+    // If a primary is currently unhealthy or more than one node is unhealthy, skip for now
+    if (unhealthyReplicasSet.has(primary) || unhealthyReplicas.length > 1) {
+      this.log(`[issueUpdateReplicaSetOp] userId=${userId} skipping reconfig.`)
+      return
+    }
 
-    this.log(`Updating Replica Set for userID ${userId} & wallet ${wallet} from old replicaSet [${primary},${secondary1},${secondary2}] due to unhealthy replica ${unhealthyReplica}`)
+    let phase = ''
+    try {
+      // Generate new replica set, excluding current replica set
+      phase = issueUpdateReplicaSetOpPhases.SELECT_CREATOR_NODES
+      const currentHealthySecondary = unhealthyReplicasSet.has(secondary1) ? secondary2 : secondary1
+
+      // make sure new secondary is not in current replica set
```

Comment:
lets rewrite this to be less confusing haha

just filtering out the current replica set from healthyPeers and then random selection should work